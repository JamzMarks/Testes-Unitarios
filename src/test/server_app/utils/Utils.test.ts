import { IncomingMessage } from "http"
import { getRequestBody } from "../../../app/server_app/utils/Utils"

describe('Utils test suit', () => {

    const requestMock = {
        on: jest.fn()
    }

    const someObj = {
        name: 'SomeName',
        age: 30,
        city: 'Paris'
    }

    const someObjAsString = JSON.stringify(someObj)

    it('should return object for valid JSON', async () => {

        requestMock.on.mockImplementation((event, cb) => {
            if(event === 'data'){
                cb(someObjAsString)
            }else{
                cb()
            }
        })

        const actual = await getRequestBody(
            requestMock as any as IncomingMessage
        )

        expect(actual).toEqual(someObj)

    })

    it('should throw error for invalid JSON', async () => {
        requestMock.on.mockImplementation((event, cb) => {
            if(event === 'data'){
                cb('a' + someObjAsString)
            }else{
                cb()
            }
        })

        await expect(getRequestBody(requestMock as any)).rejects.
            toThrowErrorMatchingSnapshot()
    })

    it('should throw error for unexpected error', async () => {
        const someError = new Error('Something went wrong')
        requestMock.on.mockImplementation((event, cb) => {
            if(event === 'error'){
                cb(someError)
            }
        })
        await expect(getRequestBody(requestMock as any)).rejects.
        toThrowErrorMatchingSnapshot()
    })
})