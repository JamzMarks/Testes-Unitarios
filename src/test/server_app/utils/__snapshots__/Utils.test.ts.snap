// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils test suit should throw error for invalid JSON 1`] = `"Unexpected token 'a', "a{"name":""... is not valid JSON"`;

exports[`Utils test suit should throw error for unexpected error 1`] = `"Something went wrong"`;
